sources/Application/Instruments/SamplePool.cpp

```diff
@@ -166,7 +166,12 @@ bool SamplePool::loadSample(const char *path) {
   WavFile *wave = WavFile::Open(path);
   if (wave) {
     wav_[count_] = wave;
-    const std::string name = wavPath.GetName();
+    std::string name = wavPath.GetName();
+    // Drop file extension
+    size_t index = name.find_last_of(".");
+    if (index < name.length() && index > 0) {
+      name.resize(index);
```

Comment:
I suggested using name.substr, because this method accepts `string::npos` seamlessly, returning the full string if the len is string::npos.

Reply:
Yeah, I didn't realise substr access string::npos.. it is a much cleaner solution, anyway that has been implemented and this should be resolved.